### PR TITLE
Add public upload and url generation for S3 files.

### PR DESCRIPTION
**Pull request by eweisner1973**
This adds two basic methods for programmatically handling files hosted via S3 that will be publicly available. One is an upload method, the other is a URL generation method.

### CODE REVIEW

**Comment by eweisner1973** [comment]
> Curious if we could extend existing functions to handle this use-case. Also, these aren't being used by the Files app, so what is the use-case for them? Some doc-comments could help clear it up.

I wondered if the belonged here as well, and wound up 

> Curious if we could extend existing functions to handle this use-case. Also, these aren't being used by the Files app, so what is the use-case for them? Some doc-comments could help clear it up.

The use case is pretty rare. Basically trying to programmatically mimic what happens when a file is uploaded via a Django admin form using a field type that supports upload_to such as models.ImageField. When those types of fields are used on a form, behind the scenes we get the file uploaded to the appropriate S3 server, and a public URL that never expires rather than a generated URL that does expire. In my case, I had a website scraper downloading images, uploading them, and using the URL data in new records.

The logic may not belong here as it doesn't strictly require or use the File model.

File: apps/file/libs.py [hunk 1]
```diff
@@ -77,3 +77,30 @@ def get_signed_url(method, params):
...
     return _client().generate_presigned_url(method + '_object', params)
+
+
+def get_public_url(file_name):
+    # Only works for files uploaded with ACL: public-read (upload_public_file)
+    return "https://s3.{0}.amazonaws.com/{1}/{2}".format(
+        settings.AWS_S3_REGION_NAME,
+        settings.AWS_STORAGE_BUCKET_NAME,
+        file_name
+    )
```
[the remark this answers is not in the document]
Comment (eweisner1973) [reply apps/file/libs.py]: Hmm, I tried this with the ExpiresIn set to 0 and wound up with a not authorized error. But I don't know that I changed the signature to config.signature_version = botocore.UNSIGNED. Will give that a whirl.

File: apps/file/libs.py [hunk 2]
```diff
@@ -77,3 +77,30 @@ def get_signed_url(method, params):
...
+
+def upload_public_file(source_file_path, file_name):
+    """
+    This method uploads a file for direct public access
+    source_file_path - A path to the file to upload
+    file_name - The identifier (key) for S3
+    """
+    mime_type = mimetypes.guess_type(file_name)[0]
+
+    _client().upload_file(
```
[the remark this answers is not in the document]
Comment (eweisner1973) [reply apps/file/libs.py]: Yah, I think I mistakenly thought file parameter was a File class. I see now:

file - A file like object/buffer

File: apps/file/libs.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (staydecent) [reply apps/file/libs.py]: Ah yeah, might not be worth the trouble. For rare cases, I think using the `get_public_url` function you defined works. Maybe could add a comment with a link to this, or the boto issue.

**Pull request merged**